src/NzbDrone.Core/Download/IDownloadClient.cs

```diff
@@ -13,5 +13,6 @@ public interface IDownloadClient : IProvider
         IEnumerable<DownloadClientItem> GetItems();
         void RemoveItem(string downloadId, bool deleteData);
         DownloadClientInfo GetStatus();
+        void SetImportedCategory(DownloadClientItem downloadClientItem);
```

Comment:
How about renaming 'SetImportCategory' to something like 'MarkItemAsImported'. This allows various download clients to behave differently depending on their individual capabilities.

Reply:
Doesn't that imply the scope of the method is broader/different than just setting the imported category? Unless the plan is to have something like 'SetImportedCategory' called under 'MarkItemAsImported' among other future features

Reply:
> Unless the plan is to have something like 'SetImportedCategory' called under 'MarkItemAsImported' among other future features

Yes. The rest of Sonarr is completely unaware of the presence of Categories.
It's the download client that decides to use a Category to filter 'sonarr' downloads. So it should also be the download client that decides to use an Import Category after successful imports. 
'MarkItemAsImported' is a good generic name for the concept, which just happens to be implemented as ImportedCategory for deluge.